session/database.rst

```diff
@@ -0,0 +1,600 @@
+            // ...
+            'session' => [
+                'handler_id' => PdoSessionHandler::class,
+            ],
+        ]);
+
+.. caution::
+
+    If the session data doesn't fit in the data column, it might get truncated
+    by the database engine. To make matters worse, when the session data gets
+    corrupted, PHP ignores the data without giving a warning.
+
+    If the application stores large amounts of session data, this problem can
+    be solved by increasing the column size (use ``BLOB`` or even ``MEDIUMBLOB``).
+    When using MySQL as the database engine, you can also enable the `strict SQL mode`_
+    to be notified when such an error happens.
+
+Configuring the Session Table and Column Names
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The table used to store sessions is called ``sessions`` by default and defines
+certain column names. You can configure these values with the second argument
+passed to the ``PdoSessionHandler`` service:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            # ...
+
+            Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler:
+                arguments:
+                    - '%env(DATABASE_URL)%'
+                    - { db_table: 'customer_session', db_id_col: 'guid' }
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler" public="false">
+                    <argument>env(DATABASE_URL)</argument>
+                    <argument type="collection">
+                        <argument key="db_table">customer_session</argument>
+                        <argument key="db_id_col">guid</argument>
+                    </argument>
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+        use Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler;
+        // ...
+
+        $container->autowire(PdoSessionHandler::class)
+            ->setArguments([
+                '%env(DATABASE_URL)%',
+                ['db_table' => 'customer_session', 'db_id_col' => 'guid'],
+            ])
+        ;
+
+These are parameters that you can configure:
+
+``db_table`` (default ``sessions``):
+    The name of the session table in your database;
+
+``db_username``: (default: ``''``)
+    The username used to connect when using the PDO configuration (it's ignored
+    when using the connection based on the ``DATABASE_URL`` env var).
```

Comment:
It is not ignored. If you pass both the `db_username` option (other than `null`) and a username in the URL, the explicit option still wins (same for `db_password`). However, in most cases, you will want to rely on the URL if you pass it.